src/System.Windows.Extensions/tests/System/Drawing/Printing/MarginsConverterTests.cs

```diff
@@ -0,0 +1,164 @@
+            object result = mc.CreateInstance(context, values);
+            Assert.NotNull(result);
+
+            Assert.IsType<Margins>(result);
+            Margins margins = result as Margins;
+            Assert.Equal(1, margins.Left);
+            Assert.Equal(2, margins.Right);
+            Assert.Equal(3, margins.Top);
+            Assert.Equal(4, margins.Bottom);
+        }
+
+        [Fact]
+        public void GetCreateInstanceSupported()
+        {
+            MarginsConverter mc = new MarginsConverter();
+            Assert.True(mc.GetCreateInstanceSupported(null));
+            Assert.True(mc.GetCreateInstanceSupported(new MyTypeDescriptorContext()));
+        }
+
+        [Fact]
+        public void ConvertFrom()
+        {
+            MarginsConverter mc = new MarginsConverter();
+            CultureInfo culture = CultureInfo.InvariantCulture;
+
+            // try once with then once without context
+            for (var context = new MyTypeDescriptorContext(); context != null; context = null)
+            {
+                if (PlatformDetection.IsFullFramework)
+                {
+                    Assert.Throws<Exception>(() => mc.ConvertFrom(context, culture, "1;2;3;4"));
+                    return;
+                }
+                object result = mc.ConvertFrom(context, culture, "1;2;3;4");
+                Assert.IsType<Margins>(result);
+                Margins margins = result as Margins;
+                Assert.Equal(1, margins.Left);
+                Assert.Equal(2, margins.Right);
+                Assert.Equal(3, margins.Top);
+                Assert.Equal(4, margins.Bottom);
+
+                Assert.Throws<NotSupportedException>(() => mc.ConvertFrom(context, null, null));
+                Assert.Throws<NotSupportedException>(() => mc.ConvertFrom(context, culture, null));
+                Assert.Throws<NotSupportedException>(() => mc.ConvertFrom(context, culture, Guid.NewGuid()));
+
+                Assert.Throws<ArgumentException>(() => mc.ConvertFrom(context, null, "wrong string format"));
+                Assert.Throws<ArgumentException>(() => mc.ConvertFrom(context, culture, "wrong string format"));
+            }
+        }
+
+        [Fact]
+        public void ConvertTo()
+        {
+            MarginsConverter mc = new MarginsConverter();
+            Guid guid = Guid.NewGuid();
+            CultureInfo culture = CultureInfo.InvariantCulture;
+            Margins margins = new Margins() { Left = 1, Right = 2, Top = 3, Bottom = 4 };
+
+            // try once with then once without context
+            for (var context = new MyTypeDescriptorContext(); context != null; context = null)
+            {
+                Assert.Equal("1;2;3;4", mc.ConvertTo(context, culture, "1;2;3;4", typeof(string)));
+
+                object converted = mc.ConvertTo(context, culture, margins, typeof(string));
+                Assert.IsType<string>(converted);
+                if (PlatformDetection.IsFullFramework)
```

Comment:
What's causing these behavior differences?

Reply:
Looking

Reply:
In case you are going to keep any of the Mono code, please make sure to put the right license on it.

Reply:
@maryamariyan specifically, to reuse Mono code you should retain all existing copyright headers on the file, and then add this:
```
// Licensed to the .NET Foundation under one or more agreements.
// See the LICENSE file in the project root for more information.
```
Note that it is missing the middle line of the header we normally use.